### ISSUE TITLE
Deck application causes error when activated and nextcloud search is used

### ISSUE THREAD

**Issue by talesam** [nextcloud/deck#2988]
Tested on version **21.0.1** of Nextcloud

1) I install the Deck app
2) Search and search for anything, the following error appears: "**An error occurred while searching for Card comments**"

![Captura de tela de 2021-04-13 20-46-39](https://user-images.githubusercontent.com/981368/114634643-6302f680-9c99-11eb-9e2b-2c64de5f2207.png)

Deactivating the Deck, the search normalizes.

```
[no app in context] Error: Doctrine\DBAL\Exception\DriverException: An exception occurred while executing a query: SQLSTATE[42883]: Undefined function: 7 ERROR:  operator does not exist: character varying = integer
LINE 1: ...c_comments" "comments" ON ("comments"."object_id" = "c"."id"...
                                                             ^
HINT:  No operator matches the given name and argument types. You might need to add explicit type casts. at <<closure>>

 0. /var/www/html/3rdparty/doctrine/dbal/src/Connection.php line 1728
    Doctrine\DBAL\Driver\API\PostgreSQL\ExceptionConverter->convert(Doctrine\DBAL\Driver\PDO\Exception {}, Doctrine\DBAL\Query {})
 1. /var/www/html/3rdparty/doctrine/dbal/src/Connection.php line 1667
    Doctrine\DBAL\Connection->handleDriverException(Doctrine\DBAL\Driver\PDO\Exception {}, Doctrine\DBAL\Query {})
 2. /var/www/html/3rdparty/doctrine/dbal/src/Connection.php line 1040
    Doctrine\DBAL\Connection->convertExceptionDuringQuery(Doctrine\DBAL\Driver\PDO\Exception {}, "SELECT \"c\".*, ... 5", ["deckCard",0,0,0,"%google%"], [2,1,1,1,2])
 3. /var/www/html/lib/private/DB/Connection.php line 226
    Doctrine\DBAL\Connection->executeQuery("SELECT \"c\".*, ... 5", ["deckCard",0,0,0,"%google%"], [2,1,1,1,2], null)
 4. /var/www/html/3rdparty/doctrine/dbal/src/Query/QueryBuilder.php line 210
    OC\DB\Connection->executeQuery("SELECT \"c\".*, ... 5", {0: "And 1 more  ... "}, {0: "And 1 more  ... 2})
 5. /var/www/html/lib/private/DB/QueryBuilder/QueryBuilder.php line 287
    Doctrine\DBAL\Query\QueryBuilder->execute()
 6. /var/www/html/custom_apps/deck/lib/Db/CardMapper.php line 351
    OC\DB\QueryBuilder\QueryBuilder->execute()
 7. /var/www/html/custom_apps/deck/lib/Service/SearchService.php line 104
    OCA\Deck\Db\CardMapper->searchComments([], OCA\Deck\Search\Query\SearchQuery {}, 5, null)
 8. /var/www/html/custom_apps/deck/lib/Search/CardCommentProvider.php line 61
    OCA\Deck\Service\SearchService->searchComments("google", 5, null)
 9. /var/www/html/lib/private/Search/SearchComposer.php line 162
    OCA\Deck\Search\CardCommentProvider->search(OC\User\User {}, OC\Search\SearchQuery {})
10. /var/www/html/core/Controller/UnifiedSearchController.php line 110
    OC\Search\SearchComposer->search(OC\User\User {}, "deck-comment", OC\Search\SearchQuery {})
11. /var/www/html/lib/private/AppFramework/Http/Dispatcher.php line 218
    OC\Core\Controller\UnifiedSearchController->search("deck-comment", "google", null, null, null, "/settings/apps/office")
12. /var/www/html/lib/private/AppFramework/Http/Dispatcher.php line 127
    OC\AppFramework\Http\Dispatcher->executeController(OC\Core\Controll ... {}, "search")
13. /var/www/html/lib/private/AppFramework/App.php line 157
    OC\AppFramework\Http\Dispatcher->dispatch(OC\Core\Controll ... {}, "search")
14. /var/www/html/lib/private/Route/Router.php line 302
    OC\AppFramework\App::main("OC\\Core\\Contr ... r", "search", OC\AppFramework\ ... {}, {providerId: "de ... "})
15. /var/www/html/ocs/v1.php line 63
    OC\Route\Router->match("/ocsapp/search/ ... h")
16. /var/www/html/ocs/v2.php line 24
    require_once("/var/www/html/ocs/v1.php")

GET /ocs/v2.php/search/providers/deck-comment/search?term=google&from=%2Fsettings%2Fapps%2Foffice
from 192.168.16.10 by admin at 2021-04-13T20:15:00+00:00
```

See: https://youtu.be/OJldN9PJQl8

**Comment by juliushaertl** [comment]
Thanks for your report. A fix is in #2989 ([`abacc9a`](https://github.com/nextcloud/deck/pull/2989/commits/abacc9a83c864ef63ed414f9a687246ca9cae1dd) to be exact)

**Comment by tcoupin** [comment]
Hi,

I have the same issue with NC 21.0.0 and Deck 1.4.2 : `An error occurred while searching for Deck` and `An error occurred while searching for Card comments`

```
{"reqId":"hSrIcEDf9sysSdo3vS88","level":3,"time":"2021-05-11T05:45:57+00:00","remoteAddr":"10.42.6.0","user":"admin","app":"no app in context","method":"GET","url":"/ocs/v2.php/search/providers/deck-comment/search?term=toto&from=%2Fsettings%2Fapps%2Finstalled%2Fdeck","message":{"Exception":"Doctrine\\DBAL\\Exception\\DriverException","Message":"An exception occurred while executing a query: SQLSTATE[42000]: Syntax error or access violation: 1055 'nextcloud_db_prod.c.title' isn't in GROUP BY","Code":1055,"Trace":[{"file":"/var/www/html/3rdparty/doctrine/dbal/src/Connection.php","line":1728,"function":"convert","class":"Doctrine\\DBAL\\Driver\\API\\MySQL\\ExceptionConverter","type":"->","args":[{"__class__":"Doctrine\\DBAL\\Driver\\PDO\\Exception"},{"__class__":"Doctrine\\DBAL\\Query"}]},{"file":"/var/www/html/3rdparty/doctrine/dbal/src/Connection.php","line":1667,"function":"handleDriverException","class":"Doctrine\\DBAL\\Connection","type":"->","args":[{"__class__":"Doctrine\\DBAL\\Driver\\PDO\\Exception"},{"__class__":"Doctrine\\DBAL\\Query"}]},{"file":"/var/www/html/3rdparty/doctrine/dbal/src/Connection.php","line":1040,"function":"convertExceptionDuringQuery","class":"Doctrine\\DBAL\\Connection","type":"->","args":[{"__class__":"Doctrine\\DBAL\\Driver\\PDO\\Exception"},"SELECT `c`.*, `comments`.`id` AS `comment_id` FROM `oc_deck_cards` `c` INNER JOIN `oc_deck_stacks` `s` ON `s`.`id` = `c`.`stack_id` INNER JOIN `oc_comments` `comments` ON (`comments`.`object_id` = `c`.`id`) AND (`comments`.`object_type` = ?) INNER JOIN `oc_deck_boards` `b` ON `b`.`id` = `s`.`board_id` WHERE (`s`.`board_id` IN (NULL)) AND (`c`.`deleted_at` = ?) AND (`s`.`deleted_at` = ?) AND (`b`.`deleted_at` = ?) AND (`comments`.`message`  COLLATE utf8mb4_general_ci LIKE ?) GROUP BY `comments`.`id`, `c`.`id` ORDER BY `comments`.`id` DESC LIMIT 5",["deckCard",0,0,0,"%toto%"],[2,1,1,1,2]]},{"file":"/var/www/html/lib/private/DB/Connection.php","line":226,"function":"executeQuery","class":"Doctrine\\DBAL\\Connection","type":"->","args":["SELECT `c`.*, `comments`.`id` AS `comment_id` FROM `oc_deck_cards` `c` INNER JOIN `oc_deck_stacks` `s` ON `s`.`id` = `c`.`stack_id` INNER JOIN `oc_comments` `comments` ON (`comments`.`object_id` = `c`.`id`) AND (`comments`.`object_type` = ?) INNER JOIN `oc_deck_boards` `b` ON `b`.`id` = `s`.`board_id` WHERE (`s`.`board_id` IN (NULL)) AND (`c`.`deleted_at` = ?) AND (`s`.`deleted_at` = ?) AND (`b`.`deleted_at` = ?) AND (`comments`.`message`  COLLATE utf8mb4_general_ci LIKE ?) GROUP BY `comments`.`id`, `c`.`id` ORDER BY `comments`.`id` DESC LIMIT 5",["deckCard",0,0,0,"%toto%"],[2,1,1,1,2],null]},{"file":"/var/www/html/3rdparty/doctrine/dbal/src/Query/QueryBuilder.php","line":210,"function":"executeQuery","class":"OC\\DB\\Connection","type":"->","args":["SELECT `c`.*, `comments`.`id` AS `comment_id` FROM `oc_deck_cards` `c` INNER JOIN `oc_deck_stacks` `s` ON `s`.`id` = `c`.`stack_id` INNER JOIN `oc_comments` `comments` ON (`comments`.`object_id` = `c`.`id`) AND (`comments`.`object_type` = :dcValue5) INNER JOIN `oc_deck_boards` `b` ON `b`.`id` = `s`.`board_id` WHERE (`s`.`board_id` IN (:dcValue1)) AND (`c`.`deleted_at` = :dcValue2) AND (`s`.`deleted_at` = :dcValue3) AND (`b`.`deleted_at` = :dcValue4) AND (`comments`.`message`  COLLATE utf8mb4_general_ci LIKE :dcValue6) GROUP BY `comments`.`id`, `c`.`id` ORDER BY `comments`.`id` DESC LIMIT 5",{"dcValue1":[],"dcValue2":0,"dcValue3":0,"dcValue4":0,"dcValue5":"deckCard","dcValue6":"%toto%"},{"dcValue1":101,"dcValue2":1,"dcValue3":1,"dcValue4":1,"dcValue5":2,"dcValue6":2}]},{"file":"/var/www/html/lib/private/DB/QueryBuilder/QueryBuilder.php","line":287,"function":"execute","class":"Doctrine\\DBAL\\Query\\QueryBuilder","type":"->","args":[]},{"file":"/var/www/html/custom_apps/deck/lib/Db/CardMapper.php","line":351,"function":"execute","class":"OC\\DB\\QueryBuilder\\QueryBuilder","type":"->","args":[]},{"file":"/var/www/html/custom_apps/deck/lib/Service/SearchService.php","line":104,"function":"searchComments","class":"OCA\\Deck\\Db\\CardMapper","type":"->","args":[[],{"__class__":"OCA\\Deck\\Search\\Query\\SearchQuery"},5,null]},{"file":"/var/www/html/custom_apps/deck/lib/Search/CardCommentProvider.php","line":61,"function":"searchComments","class":"OCA\\Deck\\Service\\SearchService","type":"->","args":["toto",5,null]},{"file":"/var/www/html/lib/private/Search/SearchComposer.php","line":162,"function":"search","class":"OCA\\Deck\\Search\\CardCommentProvider","type":"->","args":[{"__class__":"OC\\User\\User"},{"__class__":"OC\\Search\\SearchQuery"}]},{"file":"/var/www/html/core/Controller/UnifiedSearchController.php","line":110,"function":"search","class":"OC\\Search\\SearchComposer","type":"->","args":[{"__class__":"OC\\User\\User"},"deck-comment",{"__class__":"OC\\Search\\SearchQuery"}]},{"file":"/var/www/html/lib/private/AppFramework/Http/Dispatcher.php","line":218,"function":"search","class":"OC\\Core\\Controller\\UnifiedSearchController","type":"->","args":["deck-comment","toto",null,null,null,"/settings/apps/installed/deck"]},{"file":"/var/www/html/lib/private/AppFramework/Http/Dispatcher.php","line":127,"function":"executeController","class":"OC\\AppFramework\\Http\\Dispatcher","type":"->","args":[{"__class__":"OC\\Core\\Controller\\UnifiedSearchController"},"search"]},{"file":"/var/www/html/lib/private/AppFramework/App.php","line":157,"function":"dispatch","class":"OC\\AppFramework\\Http\\Dispatcher","type":"->","args":[{"__class__":"OC\\Core\\Controller\\UnifiedSearchController"},"search"]},{"file":"/var/www/html/lib/private/Route/Router.php","line":302,"function":"main","class":"OC\\AppFramework\\App","type":"::","args":["OC\\Core\\Controller\\UnifiedSearchController","search",{"__class__":"OC\\AppFramework\\DependencyInjection\\DIContainer"},{"providerId":"deck-comment","_route":"ocs.core.UnifiedSearch.search"}]},{"file":"/var/www/html/ocs/v1.php","line":63,"function":"match","class":"OC\\Route\\Router","type":"->","args":["/ocsapp/search/providers/deck-comment/search"]},{"file":"/var/www/html/ocs/v2.php","line":24,"args":["/var/www/html/ocs/v1.php"],"function":"require_once"}],"File":"/var/www/html/3rdparty/doctrine/dbal/src/Driver/API/MySQL/ExceptionConverter.php","Line":114,"Previous":{"Exception":"Doctrine\\DBAL\\Driver\\PDO\\Exception","Message":"SQLSTATE[42000]: Syntax error or access violation: 1055 'nextcloud_db_prod.c.title' isn't in GROUP BY","Code":1055,"Trace":[{"file":"/var/www/html/3rdparty/doctrine/dbal/src/Driver/PDO/Statement.php","line":84,"function":"new","class":"Doctrine\\DBAL\\Driver\\PDO\\Exception","type":"::","args":[{"errorInfo":["42000",1055,"'nextcloud_db_prod.c.title' isn't in GROUP BY"],"__class__":"PDOException"}]},{"file":"/var/www/html/3rdparty/doctrine/dbal/src/Connection.php","line":1030,"function":"execute","class":"Doctrine\\DBAL\\Driver\\PDO\\Statement","type":"->","args":[]},{"file":"/var/www/html/lib/private/DB/Connection.php","line":226,"function":"executeQuery","class":"Doctrine\\DBAL\\Connection","type":"->","args":["SELECT `c`.*, `comments`.`id` AS `comment_id` FROM `oc_deck_cards` `c` INNER JOIN `oc_deck_stacks` `s` ON `s`.`id` = `c`.`stack_id` INNER JOIN `oc_comments` `comments` ON (`comments`.`object_id` = `c`.`id`) AND (`comments`.`object_type` = ?) INNER JOIN `oc_deck_boards` `b` ON `b`.`id` = `s`.`board_id` WHERE (`s`.`board_id` IN (NULL)) AND (`c`.`deleted_at` = ?) AND (`s`.`deleted_at` = ?) AND (`b`.`deleted_at` = ?) AND (`comments`.`message`  COLLATE utf8mb4_general_ci LIKE ?) GROUP BY `comments`.`id`, `c`.`id` ORDER BY `comments`.`id` DESC LIMIT 5",["deckCard",0,0,0,"%toto%"],[2,1,1,1,2],null]},{"file":"/var/www/html/3rdparty/doctrine/dbal/src/Query/QueryBuilder.php","line":210,"function":"executeQuery","class":"OC\\DB\\Connection","type":"->","args":["SELECT `c`.*, `comments`.`id` AS `comment_id` FROM `oc_deck_cards` `c` INNER JOIN `oc_deck_stacks` `s` ON `s`.`id` = `c`.`stack_id` INNER JOIN `oc_comments` `comments` ON (`comments`.`object_id` = `c`.`id`) AND (`comments`.`object_type` = :dcValue5) INNER JOIN `oc_deck_boards` `b` ON `b`.`id` = `s`.`board_id` WHERE (`s`.`board_id` IN (:dcValue1)) AND (`c`.`deleted_at` = :dcValue2) AND (`s{"reqId":"bi3yMUBO95KRQ1CL0c3d","level":0,"time":"2021-05-11T05:45:57+00:00","remoteAddr":"10.42.6.0","user":"admin","app":"group_everyone","method":"GET","url":"/ocs/v2.php/search/providers/settings_apps/search?term=toto&from=%2Fsettings%2Fapps%2Finstalled%2Fdeck","message":"/appinfo/app.php is deprecated, use \\OCP\\AppFramework\\Bootstrap\\IBootstrap on the application class instead.","userAgent":"Mozilla/5.0 (Macintosh; Intel Mac OS X 10.15; rv:88.0) Gecko/20100101 Firefox/88.0","version":"21.0.1.1"}
```
